### ISSUE TITLE
New branch structure

### ISSUE THREAD

**Issue by sgrossberndt** [VDVde/OJP#435]
# Current state

- `master`
- `changes_for_v1.1`
- `trip_monitoring` example for a feature branch
- `typos-2023-11-07` example for a bugfix branch

`changes_for_v1.1` is deployed to https://vdvde.github.io/OJP/index.html

# Proposed state

- `main` contains the most recent released state (former `master`)
- `develop` contains the most recent agreed and merged development state (former `changes_for_v1.1`)
- `release/1.0` contains the most recent released state of the major version 1.0
- `release/2.0` contains the most recent released state of the major version 2.0
- `feature/trip_monitoring` features are created as `feature/*` branches and merged into `develop`
- `bugfix/typos-2023-11-07` bugfixes are created as `bugfix/*` branches and merged into `develop` and, if necessary for a published release, into release/v2.0

https://vdvde.github.io/OJP/index.html is a starting point file containing JavaScript which checks for subdirectories and creates links to them
https://vdvde.github.io/OJP/main/index.html contains the documentation rendered by the main branch, updated on push to main
https://vdvde.github.io/OJP/develop/index.html contains the documentation rendered by the develop branch, updated on push to develop
https://vdvde.github.io/OJP/release/v2.0.0/index.html contains the documentation rendered of tag v2.0.0, updated on create/published of [release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)
https://vdvde.github.io/OJP/release/v2.0.1/index.html contains the documentation rendered of tag v2.0.1, updated on create/published of [release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)

https://github.com/MicrosoftDocs/azure-devops-docs/blob/main/docs/repos/git/git-branching-guidance.md#manage-releases

**Comment by sgrossberndt** [comment]
The default github-pages-upload action https://github.com/actions/deploy-pages we are usingright now does not allow to specify an upload folder. This made the proposed documentation structure impossible or at least very hard to achieve, but switching to 
https://github.com/marketplace/actions/deploy-to-github-pages will make it easy as it allows to define an upload directory

**Comment by ue71603** [comment]
Sounds fine with me. @skinkie  what do you think?

**Comment by sgrossberndt** [comment]
I also had a look regarding getting changes_for_v.1.1 to main.

Up to version v1.0.1, which was also made a release at https://github.com/VDVde/OJP/releases things are fine.

A version v1.0.2 was tagged which solves some xmllint issues, but apparently not all of them as a v1.0.3 was released then, which not only added another fix ["Fix schemaLocation towards siri_reference-v2.0.xsd (#211)"](https://github.com/VDVde/OJP/commit/c6eabe8fca0f606a8ae6648649270ce08a7874f7) but also did several unfortunate additions to the master branch which make merging changes_for_v1.1 / 2.0 even more difficult, they also contain several problems:

- [d5550b9](https://github.com/VDVde/OJP/commit/d5550b98a4af6b4f0206fcc23f3baf7bceee1ff4) add a travis integration that lints XML files with one space (I don't know why this was added to master at all in that state as Travis integration was already done in changes_for_v1.1 back then) 

- [a39103a](https://github.com/VDVde/OJP/commit/a39103a272ea6c42dc2cf7c76ad02da216ef8729) add a commit "Proposal for a TripStatusGroup" which should only have been added to v1.1. It was added to changes_for_v1.1 later then and there were changes to the pull request before even which renamed the element "Unable" to ["Infeasible"](https://github.com/VDVde/OJP/pull/116/files#diff-231d41fa75aa5fbc0757e95a629f845d7143dbb82f7943636c37fd0e3259729bR950) [as discussed here](https://github.com/VDVde/OJP/commit/1cacd36a420d31546676bbf11090b458d8ed15aa#diff-05dc91f794abe0531d8ba05d516489cefa008800240c42747b5df62802dc31e5R955) which is a conflict now.
 [As discussed here](https://github.com/VDVde/OJP/pull/38) I implemented the proposed "option A" back then by rewriting commit history, but apparently this has been overwritten in the meantime..

- [2c5d18b](https://github.com/VDVde/OJP/commit/2c5d18be0e45e525858c33ded92fcc9f7f524b85) add a Travis CI update commit which should have been squash-merged

Usually one would want to have a complete succession of commits for all releases, but in this case I'd like to do it differently:
All additions to master after v1.0.1 are either already part of changes_for_v1.1 or not necessary for it. Merging changes_for_v1.1. into master leads to merge conflicts which are difficult to resolve. In order to make things easier I would like to ignore all additions to the master branch after v1.0.1: the current master branch becomes releases/1.0, main starts at v1.0.1 and changes_for_v1.1 becomes develop.

**Comment by skinkie** [comment]
I just hope that the generation of the documentation can be 'stacked'. My understandig is that the generation happens once the pull request comes in, _not_ after it is merged.

**Comment by sgrossberndt** [comment]
As I wrote in https://github.com/VDVde/OJP/issues/435#issuecomment-1805879263 I am sure stacking is possible when switching from the default github-pages-upload action to https://github.com/marketplace/actions/deploy-to-github-pages

You are correct that my proposal is missing the documentation for the feature/bugfix branches, I will update the proposal accordingly.

**Comment by sgrossberndt** [comment]
I just updated the proposal.

**Comment by skinkie** [comment]
Sonuds good to me.

**Comment by ue71603** [comment]
* bugfixes are added to main too? I think yes.
* release x.0 branches. Will there be x.x branches too? or would 1.0.1 be in the 1.0 branch. Should we label them as releases anyhow? I think only x.x and certainly not x.x.x so we could have a release/1.1, but we would not have a release 1.1.1 or do you prefer to have them.
* About the merging. I can agree that we do base release/2.0 on 1.0.1. But will release/1.0 still be 1.0.3 (which I would prefer).

**Comment by herlitze** [comment]
@sgrossberndt thank you for your thoughts on this and the explanation! I approve your suggestion.

**Comment by sgrossberndt** [comment]
@ue71603 

> * bugfixes are added to main too? I think yes.

Yes, once they are part of a released version for the latest major version

>     * release x.0 branches. Will there be x.x branches too? or would 1.0.1 be in the 1.0 branch. Should we label them as releases anyhow? I think only x.x and certainly not x.x.x so we could have a release/1.1, but we would not have a release 1.1.1 or do you prefer to have them.

This depends on how granular later releases may happen. Would we release a 1.0.99 after having released a 1.1.0? If yes, it could make sense. But it would not be mandatory to have them.

>     * About the merging. I can agree that we do base release/2.0 on 1.0.1. But will release/1.0 still be 1.0.3 (which I would prefer).

Yes, that was apparently ambiguous: I only want to base release/2.0 on v1.0.1, release/1.0 is based on v1.0.3.
